File: Frontend/src/interfaces/transaction-model.ts
import { TranStatus } from "./TranStatus";
import { TranType } from "./TranType";

export interface TransactionModel {
    id:string | null;
    transactionAmount: number | null;
    sourceClientBankAccountId: string | null;
    destinationClientBankAccountId: string | null;
    sourceClientBankAccount: string | null;
    destinationClientBankAccount: string | null;
    transactionType: TranType;
    transactionDate: Date | null;
    transactionDateUpdated : Date | null;
}

File: Backend/Domain/Models/TransactionModel.cs
using Domain.Enums;
namespace Domain.Models
{
    public class TransactionModel
    {
        public Guid? Id { get; set; }
        public Guid? UserId { get; set; } = default!;
        public string? UserName { get; set; } = default!;
        public decimal TransactionAmount { get; set; }
        public Guid? SourceClientBankAccountId { get; set; }
        public Guid? DestinationClientBankAccountId { get; set; }
        public string? SourceClientBankAccount { get; set; }
        public string? DestinationClientBankAccount { get; set; }
        public TranType TransactionType { get; set; }
        public DateTime? TransactionDate { get; set; }
        public DateTime? TransactionDateUpdated { get; set; }
    }
}
